src/app/my-account/my-account.component.ts

```diff
@@ -66,8 +66,15 @@ export class MyAccountComponent implements OnInit {
   }
 
   deleteMethod(method: PaymentMethod) {
-    this.donationService.deleteStripePaymentMethod(this.person, method, this.jwtAsString()).
-    subscribe(this.loadPaymentMethods)
+    this.paymentMethods = undefined;
+
+    this.donationService.deleteStripePaymentMethod(this.person, method, this.jwtAsString()).subscribe(
+      this.loadPaymentMethods.bind(this),
+      error => {
+        this.loadPaymentMethods.bind(this)()
+        alert(error.error.error)
+      }
```

Comment:
then whether the backend does what we ask of it or returns an error, we update the list of payment methods. If we had an error we display it using the native alert function.